app/models/authenticator/httpd.rb

```diff
@@ -97,6 +96,15 @@ def lookup_by_identity(username, request = nil)
 
     private
 
+    def normalize_username(username)
+      username = super(username)
+
+      # Fix encoding of username from X-Remote-User request header
+      username = username.force_encoding("UTF-8") if username.encoding == Encoding::ASCII_8BIT
```

Comment:
Same as above, if this is returning the value, can we return a copy of the passed in `username`?  I don't know if anything would expect `ASCII-8BIT`.